src/main/java/com/google/api/codegen/config/DiscoveryMethodModel.java

```diff
@@ -80,6 +83,13 @@ public DiscoveryField getInputField(String fieldName) {
         && DiscoGapicParser.getMethodInputName(method).toLowerCamel().equals(fieldName)) {
       return DiscoveryField.create(method.request(), apiModel);
     }
+    if (hasExtraFieldMask() && DiscoveryMethodTransformer.FIELDMASK_STRING.equals(fieldName)) {
+      if (fieldMaskField == null) {
+        fieldMaskField = createFieldMaskField();
+      }
+      return fieldMaskField;
+    }
```

Comment:
Would it be appropriate to add
```java
if (hasExtraFieldMask()) fieldMaskField = createFieldmaskField();
```
to the constructor? The field can be final and we make the decision in one place.

Reply:
oh sure
done.